Upgrade ubuntu to 22.04 second try

tlbdk opened the pull request:
Move Ubuntu 18.04 source setup to common image
Install libssl1.1 so we support common tools build for older ubuntu versions
Workaround issue with newer libssl and older node versions
Also tag images with node version and commit sha

tlbdk commented:
https://app.shortcut.com/connectedcars/story/82398/mysqldump-fails-because-libssl-so-1-1-is-not-available-in-the-final-image

tlbdk commented:
https://app.shortcut.com/connectedcars/story/82414/openssl-breaks-nodejs-crypto-library

tlbdk commented:
https://app.shortcut.com/connectedcars/story/82400/investigate-ways-we-can-roll-back-easily-if-base-image-breaks

chribsen commented:
Is the crypto issue with nodejs also fixed in this PR?

mathiastj commented:
I think we need to bump node to 16.17 to fix the crypto issue: https://github.com/nodejs/node/issues/42919

chribsen commented on README.md:
````diff
@@ -71,5 +71,16 @@ echo "<encypted key>" | base64 -D | gcloud kms decrypt --plaintext-file=- --ciph
+
+``` bash
+export OLD_SHA=abcd1234
+for NODE_VERSION in 18.7.0 16.16.0 14.20.0 12.22.12; do
````
@mathiastj's comment
https://github.com/connectedcars/docker-node/pull/71#issuecomment-1253355736

tlbdk commented:
> nodejs/node#42919

I just tested the example script on v16.16.0 and it works:

``` bash
root@b436cdf917fb:/# node test.js 
ArrayBuffer {
  [Uint8Contents]: <fe fe 48 6a c6 0b ef b4 ba ab 79 03 bf 59 e3 7a 04 ab 69 c6 9d cb 62 6c de fe bc b9 f0 04 28 1c 7c 1f 12 41 e7 47 fb 38 ce ed 87 97 f7 01 a1 4a ad 18 d6 57 f5 c1 5b f0 1b 4a 41 92 78 b9 fd 54 7a 26 a9 a6 7f d0 2d df e8 18 ed a6 13 a7 9d e1 c0 29 17 eb ec d0 fc 58 92 53 d5 55 66 3f 16 e9 f5 cb 04 fd ... 12245 more bytes>,
  byteLength: 12345
}
done.
root@b436cdf917fb:/# node --version
v16.16.0
```

I would prefer to do the node upgrade in a separate pull.

tlbdk commented:
> nodejs/node#42919

I just tested the example script on v16.16.0 and it works:

``` bash
root@b436cdf917fb:/# node test.js 
ArrayBuffer {
  [Uint8Contents]: <fe fe 48 6a c6 0b ef b4 ba ab 79 03 bf 59 e3 7a 04 ab 69 c6 9d cb 62 6c de fe bc b9 f0 04 28 1c 7c 1f 12 41 e7 47 fb 38 ce ed 87 97 f7 01 a1 4a ad 18 d6 57 f5 c1 5b f0 1b 4a 41 92 78 b9 fd 54 7a 26 a9 a6 7f d0 2d df e8 18 ed a6 13 a7 9d e1 c0 29 17 eb ec d0 fc 58 92 53 d5 55 66 3f 16 e9 f5 cb 04 fd ... 12245 more bytes>,
  byteLength: 12345
}
done.
root@b436cdf917fb:/# node --version
v16.16.0
```

I would prefer to do the node upgrade in a separate pull.

tlbdk commented:
> Is the crypto issue with nodejs also fixed in this PR?

It was another crypto issue you had, and it was because node 12.x was used in the image, node 12.x is out of support. I added the fix for it anyway to the image.

chribsen reviewed (approved):
Looks good, but can you wait with merging until Monday?

tlbdk commented:
> Looks good, but can you wait with merging until Monday?

yes, will do
